File: aula03-tpc/PontoDll/Ponto.cpp
// Ponto.cpp : Defines the exported functions by Ponto.dll
//

#include "Ponto.h"
#include <math.h>
#include <iostream>

/*
 * This is the constructor of a class that has been exported.
 * see Ponto.h for the class definition
 */
Ponto::Ponto(int x, int y){
	this->_x = x;
	this->_y = y;
	this->_z = 0;
}

double Ponto::getModule() {	 
	return sqrt((double)_x*_x + _y*_y + _z*_z);
}

void Ponto::Print(){
	printf("SUPER: (x = %d, y = %d, z=%d)\n", _x, _y, _z);
}

File: aula10/Ponto.cs
using System;


class Ponto {

    private int _y;

    public Ponto(int x, int y) {
        this.X = x;
        this.Y = y;
    }

    public int X
    {
        get;
        set;
    }
    public int Y
    {
        get { return _y; }
        set { this._y = value; }
    }

    public double Module {
        get {
            return Math.Sqrt(X*X + Y*Y);
        }
    }

    public override String ToString() { return String.Format("({0}, {1})", X, Y); }
}

class Program
{

    static void Main()
    {
        Ponto p1 = new Ponto(5, 7);

       

    }
}